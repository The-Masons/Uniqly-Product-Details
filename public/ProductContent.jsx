import React from 'react';

const starImages = [
  "https://i.imgur.com/BWTowvx.gif",
  "https://i.imgur.com/VZ89RCE.png",
  "https://i.imgur.com/zd4DwiU.png",
  "https://i.imgur.com/FvECFC4.png",
  "https://i.imgur.com/5Rb5yY3.png",
  "https://i.imgur.com/DYQgYtC.png"
]

class ProductOverview extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const prod = this.props.product;
    const colors = this.props.colors.map((color, index) => {
      return <Color color={color.name} key={index} img={color.img_url}/>
    })
    let stars;
    let reviewText;
    if(prod.reviewcount > 0){
      stars = Math.floor(prod.reviewscore);
      reviewText = prod.reviewcount + " reviews";
    } else {
      stars = 0;
      reviewText = "write the first review";
    }
    return (
      <div className="product-content">
        <div className="product-actions">
          <div className="social-sharing">
            <span className="share-button">Share</span>
            <span className="button-divider">|</span>
            <span className="save-button">Save</span>
          </div>
        </div>
          <div className="product-name">{prod.product_name}</div>
          <div className="sku">sku# {prod.sku}</div>
          <div className="review-parent">
            <div className="reviews_img">
              <img src={starImages[stars]}/>
            </div> 
            <span className="review-text">{reviewText}</span>
          </div>
          <div className="product-price">${prod.price}.99</div>
          <div className="color-parent">
            <span className="color-label">Color:</span> <span className="current-color">{prod.color}</span>
              <div className="product-colors">
                {colors}
              </div>
          </div>
          <div className="size-chart-section">
            <div className="size-chart-parent">
            <img src="https://i.imgur.com/ntZVBxZ.png" />
              <div className="size-chart-text">SIZE CHART</div>
            </div>
          </div>
          <div id="quick-add-app"></div>
      </div>
    );
  }
}

export default ProductOverview;

class Color extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="product-color"/>
    );
  }
}