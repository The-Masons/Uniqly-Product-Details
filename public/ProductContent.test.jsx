import React from 'react';
import ProductContent from './ProductContent.jsx';

const product = {
  description: `A 100% Supima® cotton T-shirt thick enough to wear on its own or for layering.
  The luxurious jersey material uses 100% Supima® cotton for a smooth feel, natural brightness and beautiful colors.
  A smooth silhouette from the bust down to the waist, with sleek shoulders and armholes.
  A narrower seam fold and stitching for a more delicate neckline.
  With a flattering neckline.`,
  material: `100% cotton
  Machine washable
  Imported  `,
  product_name: "Women JWA EXTRA FINE COTTON SHIRT STRIPE LONG-SLEEVE DRESS",
  sku: "AOJ23F5",
  color: "43 Periwinkle",
  price: '9',
  reviewscore: 4.2,
  reviewcount: 8
}

const colors = ["https://upload.wikimedia.org/wikipedia/commons/thumb/2/25/Red.svg/2000px-Red.svg.png", "https://vignette.wikia.nocookie.net/joke-battles/images/0/0e/Green.jpg/revision/latest?cb=20170111231844"]

describe('ProductContent', () => {
  test('should render with the correct items', () => {
    const productContent = mount(
      <ProductContent
        product={product} 
        colors={colors}
        />
      );
    expect(productContent).toMatchSnapshot();
  });
  
});