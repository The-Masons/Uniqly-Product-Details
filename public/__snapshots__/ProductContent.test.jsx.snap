// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductContent should render with the correct items 1`] = `
<ProductOverview
  colors={
    Array [
      "https://upload.wikimedia.org/wikipedia/commons/thumb/2/25/Red.svg/2000px-Red.svg.png",
      "https://vignette.wikia.nocookie.net/joke-battles/images/0/0e/Green.jpg/revision/latest?cb=20170111231844",
    ]
  }
  product={
    Object {
      "color": "43 Periwinkle",
      "description": "A 100% Supima® cotton T-shirt thick enough to wear on its own or for layering.
  The luxurious jersey material uses 100% Supima® cotton for a smooth feel, natural brightness and beautiful colors.
  A smooth silhouette from the bust down to the waist, with sleek shoulders and armholes.
  A narrower seam fold and stitching for a more delicate neckline.
  With a flattering neckline.",
      "material": "100% cotton
  Machine washable
  Imported  ",
      "price": "9",
      "product_name": "Women JWA EXTRA FINE COTTON SHIRT STRIPE LONG-SLEEVE DRESS",
      "reviewcount": 8,
      "reviewscore": 4.2,
      "sku": "AOJ23F5",
    }
  }
>
  <div
    className="product-content"
  >
    <div
      className="product-actions"
    >
      <div
        className="social-sharing"
      >
        <span
          className="share-button"
        >
          Share
        </span>
        <span
          className="button-divider"
        >
          |
        </span>
        <span
          className="save-button"
        >
          Save
        </span>
      </div>
    </div>
    <div
      className="product-name"
    >
      Women JWA EXTRA FINE COTTON SHIRT STRIPE LONG-SLEEVE DRESS
    </div>
    <div
      className="sku"
    >
      sku# 
      AOJ23F5
    </div>
    <div
      className="review-parent"
    >
      <div
        className="reviews_img"
      >
        <img
          src="stars_4.png"
        />
      </div>
      <span
        className="review-text"
      >
        8 reviews
      </span>
    </div>
    <div
      className="product-price"
    >
      $
      9
      .99
    </div>
    <div
      className="color-parent"
    >
      <span
        className="color-label"
      >
        Color:
      </span>
       
      <span
        className="current-color"
      >
        43 Periwinkle
      </span>
      <div
        className="product-colors"
      >
        <Color
          key="0"
        >
          <div
            className="product-color"
          />
        </Color>
        <Color
          key="1"
        >
          <div
            className="product-color"
          />
        </Color>
      </div>
    </div>
    <div
      className="size-chart-section"
    >
      <div
        className="size-chart-parent"
      >
        <img
          src="size_chart.png"
        />
        <div
          className="size-chart-text"
        >
          SIZE CHART
        </div>
      </div>
    </div>
  </div>
</ProductOverview>
`;
